File: 45.jump-game-ii/solution.cpp
class Solution {
public:
	int jump(vector<int>& nums) {
		int *result = new int[nums.size()];
		memset(result, 0, nums.size() * sizeof(int));
		int p = 1;
		for (int i = 0; i < nums.size(); i++)
		{
			for (;p <= i + nums[i] && p < nums.size(); p++)
				result[p] = result[i] + 1;
			if (p >= nums.size())
				break;
		}
		int ans = result[nums.size() - 1];
		delete[] result;
		return ans;
	}
};

File: 45.jump-game-ii/solution.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        l = [0] * len(nums)
        k = 1
        for i in range(len(nums) - 1):
            for j in range(max(i + 1, k), min(i + nums[i] + 1, len(nums))):
                l[j] = l[i] + 1
                k = j + 1
        return l[len(nums) - 1]
